File: app/src/main/java/de/sneakpeek/data/MovieQuery.java
package de.sneakpeek.data;

import android.os.Parcel;
import android.os.Parcelable;

import com.squareup.moshi.Json;

import java.util.List;

public class MovieQuery implements Parcelable {

    public final static Parcelable.Creator<MovieQuery> CREATOR = new Creator<MovieQuery>() {

        public MovieQuery createFromParcel(Parcel in) {
            MovieQuery instance = new MovieQuery();
            instance.page = ((int) in.readValue((int.class.getClassLoader())));
            in.readList(instance.results, (de.sneakpeek.data.MovieQueryResult.class.getClassLoader()));
            instance.totalResults = ((int) in.readValue((int.class.getClassLoader())));
            instance.totalPages = ((int) in.readValue((int.class.getClassLoader())));
            return instance;
        }

        public MovieQuery[] newArray(int size) {
            return (new MovieQuery[size]);
        }

    };

    @Json(name = "page") public int page;
    @Json(name = "results") public List<MovieQueryResult> results = null;
    @Json(name = "total_results") public int totalResults;
    @Json(name = "total_pages") public int totalPages;

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(page);
        dest.writeList(results);
        dest.writeValue(totalResults);
        dest.writeValue(totalPages);
    }

    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/de/sneakpeek/data/MovieQuery.kt
package de.sneakpeek.data

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class MovieQuery(@Expose @SerializedName("page") var page: Int = 0,
                 @Expose @SerializedName("results") var results: List<MovieQueryResult>? = null,
                 @Expose @SerializedName("total_results") var totalResults: Int = 0,
                 @Expose @SerializedName("total_pages") var totalPages: Int = 0) : Parcelable {

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<MovieQuery> = object : Parcelable.Creator<MovieQuery> {
            override fun createFromParcel(source: Parcel): MovieQuery = MovieQuery(source)
            override fun newArray(size: Int): Array<MovieQuery?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readInt(),
            source.createTypedArrayList(MovieQueryResult.CREATOR),
            source.readInt(),
            source.readInt()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(page)
        dest.writeTypedList(results)
        dest.writeInt(totalResults)
        dest.writeInt(totalPages)
    }
}